src/openassetio-core/src/hostApi/Manager.cpp

```diff
@@ -323,6 +323,9 @@ void Manager::preflight(const EntityReferences &entityReferences,
                         const ContextConstPtr &context,
                         const PreflightSuccessCallback &successCallback,
                         const BatchElementErrorCallback &errorCallback) {
+  if (entityReferences.size() != entityTraitsDatas.size()) {
+    throw std::out_of_range{"Parameter lists must be of the same length"};
```

Comment:
[Consider] It might be friendly to have the sizes of the lists in the exception message.

Reply:
Updated error messages in f3fe18d4

Reply:
My kindom for `std::format`!. :)